File: sdts-backend/sdts-backend/src/main/java/com/sdts_backend/service/TaskService.java
package com.sdts_backend.service;

import java.time.LocalDateTime;
import java.util.List;
import com.sdts_backend.entity.Task;
import com.sdts_backend.entity.Priority;
import com.sdts_backend.entity.Status;

public interface TaskService {
	Task createTask(String title,String description, LocalDateTime dueDate, Priority priority, Long userId);
	List<Task> findTaskByUser(Long userId);
	Task updateTask(Long taskId, Task updatedTask);
	void deleteTask(Long taskId);
	void markTaskCompleted(Long taskId);
	List<Task> findTaskByDueDate(LocalDateTime dueDate);
	List<Task> findByStatus(Status status);
	List<Task> findByPriority(Priority priority);
	
}

File: client/src/services/taskService.js
import API from "../api/axios";

export const createTask = (task) => {
  return API.post("/tasks", task);
};

export const getTasksByUser = (userId) => {
  return API.get(`/tasks/user/${userId}`);
};

export const updateTask = (taskId, task) => {
  return API.put(`/tasks/${taskId}`, task);
};

export const markTaskCompleted = (taskId) => {
  return API.patch(`/tasks/${taskId}/COMPLETE`);
};

export const deleteTask = (taskId) => {
  return API.delete(`/tasks/${taskId}`);
};

export const getTasksByPriority = (priority) => {
  return API.get(`/tasks/priority/${priority}`);
};

export const getTasksByStatus = (status) => {
  return API.get(`/tasks/status/${status}`);
};

export const getTasksByDueDate = (dueDate) => {
  return API.get(`/tasks/due?dueDate=${dueDate}`);
};

